ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/Assertions.scala

```diff
@@ -189,6 +201,74 @@ object Assertions {
     )
   }
 
+  def assertCompletionIsDefined(optCompletion: Option[Completion]): Completion = {
+    assert(optCompletion.isDefined, "No completion has been produced")
```

Comment:
`assertSingleton` requires converting the `Option[T]` to a `List[T]`. I think this would benefit from being generalized to `assertDefined(Option[T]): T`, as you suggest.

Reply:
Yes and no, as the error message will suffer. Any idea how to preserve the same amount of information, @nicu-da / @SamirTalwar-DA?

Reply:
Pass the error message in?

Reply:
Not a huge fan of it but can do it.